libs/core/synchronization/include/hpx/synchronization/async_rw_mutex.hpp

```diff
@@ -218,12 +223,17 @@ namespace hpx { namespace experimental {
             async_rw_mutex_access_wrapper& operator=(
                 async_rw_mutex_access_wrapper const&) = default;
 
-            operator ReadWriteT&()
+            ReadWriteT& get()
```

Comment:
I assume that this function will return a non-const reference, will it? Am I right in my assumption, that the wrapper is 'valid' only while accessing (writing) the value is safe? Even then, is there a way to 'cheat' the API and to hold on to the returned reference even until after the lock has been released?

Reply:
Yes, this one will be non-const. You can definitely hold on to a reference even after the wrapper is gone, but it'll be an invalid reference since the value it points to will be moved once the wrapper (and the other wrappers pointing to the same value in the read-only case) goes out of scope. I don't know if that exact wording was intentional, but I'd flip your sentence and say: "Accessing the value is safe only while the wrapper is valid (moves are fine, copies increase the reference count)".

One more change I might do here is make the read-write wrapper non-copyable.

Reply:
Done, the read-write wrappers are now non-copyable. I've had to rebase on #5274 though, because it contains important fixes to on (on captures arguments by copy, rather than forwarding them correctly, on master; this is fixed in #5274).